### ISSUE TITLE
how do you train 9x192 nets

### ISSUE THREAD

**Issue by l1t1** [CuriosAI/sai#67]
do you use 6x128 self-play games to train them?
I notice [bdc341c6](http://sai.unich.it/networks/bdc341c6383b43836681d303fc942d27a945a33e64eeef8213fa41e4f7f634da.gz) and [d4cb44f0](http://sai.unich.it/networks/d4cb44f0dd488ede496598bc1355636e095dfeb26b393fcce8d2bae7436458ee.gz) show 4.7k or 149k training games respectively.

**Comment by Vandertic** [comment]
Yes, we use 6x18 games to train 9x192 nets, with a dedicated server. The difficulty is that we cannot filter the best net in each generation as we are doing with promotion matches of the online procedure, and it is absolutely non-obvious how many steps to use for each generation.

After a first naive attempt that failed spectacularly with networks which were weak and overfitted, now, after some considerations based on AlphaZero parameters, we are doing a second attempt that seems to be performing better. We are now at generation 48 (g030-98b0) and going on fast, at a pace of two generations per hour, so we will be finished in 40 hour or so. Every 8 hour / 16 generations I will put up one match like 98b04fd5 VS 8c85ba85, that is 9x192 against one random (not best) 6x128 network of the previous generation. They show as green triangles on the plot.

**Comment by Vandertic** [comment]
> what was the reason of overfitting ? batch size ?

The reason for the overfitting was the number of steps per generation (12000 steps was too high) possibly together with a learning rate a bit too low in the first generations (where it was 0.001).
Now we are training with 6000 steps per generation, with learning rate and window size copied, generation by generation, from the original 6x128 training. Batch size is always the default in the LZ tensorflow training script, that is 512 virtual size, made up with four size 128 mini-batches which are GPU-parallelized.
In particular, learning rate was an enormous 0.05 from generation 1 to 91, then dropped to 0.01, 0.004, 0.002, 0.001, 0.0005 every 4 generation, at 92, 96, 100, 104, 108. Rate was again dropped to 0.00025 recently, at generation 134.

**Comment by l1t1** [comment]
shall we cut down the visits of self-play from 2400 to smaller

**Comment by Vandertic** [comment]
I am not sure, but probably not. Games are slightly slower, but stronger.

**Comment by l1t1** [comment]
promote match have been  cut down, but panel match are still too much

**Comment by kennyfs** [comment]
This issue should be closed.
